src/generator/type_inference_visitor.ts

```diff
@@ -159,18 +171,7 @@ class TypeInferenceVisitor extends Visitor {
         if (isIdentifier(node.left)) {
             this.assignToIdentifier(node);
         } else if (isMemberExpression(node.left)) {
-            this.assignToArray(node);
-        }
-    }
-
-    private assignToArray(node: AssignmentExpression) {
-        const {left: memberExpression, right: expression} = node;
-
-        if (isMemberExpression(memberExpression)) {
-            super.visit(memberExpression);
-
-            const rightSideType = this.getTypeOfExpression(expression);
-            this.expressionTypes.set(memberExpression, rightSideType);
+            super.visit(node.left);
```

Comment:
There is a check missing here. If you have an int array and make a shorthand assignment to a double value it should throw an error (use the `checkIfTypeIsUnchanged` functionality).
Please fix this error and add a test for this behavior.